plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx

```diff
@@ -66,161 +63,171 @@ export const NotificationsTable = ({
   totalCount,
 }: NotificationsTableProps) => {
   const notificationsApi = useApi(notificationsApiRef);
+  const [selectedNotifications, setSelectedNotifications] = React.useState(
+    new Set<Notification['id']>(),
+  );
+
+  const onNotificationsSelectChange = React.useCallback(
+    (ids: Notification['id'][], checked: boolean) => {
+      let newSelect: Set<Notification['id']>;
+      if (checked) {
+        newSelect = new Set([...selectedNotifications, ...ids]);
+      } else {
+        newSelect = new Set(selectedNotifications);
+        ids.forEach(id => newSelect.delete(id));
+      }
+      setSelectedNotifications(newSelect);
+    },
+    [selectedNotifications, setSelectedNotifications],
+  );
 
   const onSwitchReadStatus = React.useCallback(
-    (notification: Notification) => {
+    (ids: Notification['id'][], newStatus: boolean) => {
       notificationsApi
         .updateNotifications({
-          ids: [notification.id],
-          read: !notification.read,
+          ids,
+          read: newStatus,
         })
-        .then(() => onUpdate());
+        .then(onUpdate);
     },
     [notificationsApi, onUpdate],
   );
 
   const onSwitchSavedStatus = React.useCallback(
-    (notification: Notification) => {
+    (ids: Notification['id'][], newStatus: boolean) => {
       notificationsApi
         .updateNotifications({
-          ids: [notification.id],
-          saved: !notification.saved,
+          ids,
+          saved: newStatus,
         })
-        .then(() => onUpdate());
+        .then(onUpdate);
     },
     [notificationsApi, onUpdate],
   );
 
-  const throttledContainsTextHandler = useMemo(
+  const throttledContainsTextHandler = React.useMemo(
     () => throttle(setContainsText, ThrottleDelayMs),
     [setContainsText],
   );
 
+  React.useEffect(() => {
```

Comment:
We generally prefer direct imports, i.e. `useEffect`, rather than using the `React` namespace